ducktape/tests/loader.py

```diff
@@ -32,11 +34,22 @@ class LoaderException(Exception):
 
 class TestInfo(object):
     """Helper class used to wrap discovered test information"""
-    def __init__(self, module=None, cls=None, function=None, injected_args=None):
+
+    @staticmethod
+    def from_test_info(t_info):
+        """Create a copy"""
+        return TestInfo(t_info.module, t_info.cls, t_info.function, t_info.injected_args, t_info.regression_variable_selector)
+
+    def __init__(self, module=None, cls=None, function=None, injected_args=None, regression_variable_selector=None):
```

Comment:
Just a thought that I haven't looked thoroughly into the implications of, but if we're going to end up with "fake" tests like these ones for regressions, it might make sense to make some of them subclasses of `TestInfo` instead of continuing to add parameters to the constructor here with a bunch of conditionals in the class. This works ok for now as long as we only have a few ways to customize it (injected args, regression_variable_selector), but it gets messy when scaled up.